rj_gameplay/rj_gameplay/action/receive.py

```diff
@@ -28,9 +29,12 @@ def tick(self, intent) -> None:
         return intent
 
     def is_done(self, world_state) -> bool:
+        if world_state.our_robots[self.robot_id].has_ball_sense:
+            self.has_ball_ticks += 1
+        print(self.has_ball_ticks)
         if self.robot_id is None:
             return False
         #TODO: Use local params for this threshold
-        if world_state.our_robots[self.robot_id].has_ball_sense or np.linalg.norm(world_state.ball.vel) < SETTLE_BALL_SPEED_THRESHOLD:
+        if self.has_ball_ticks >= 15 or np.linalg.norm(world_state.ball.vel) < SETTLE_BALL_SPEED_THRESHOLD:
```

Comment:
I think you should reset has_ball_ticks here

Reply:
Ah yeah forgot about that lol